modules/apps/transfer/keeper/grpc_query_test.go

```diff
@@ -140,3 +140,74 @@ func (suite *KeeperTestSuite) TestQueryParams() {
 	res, _ := suite.queryClient.Params(ctx, &types.QueryParamsRequest{})
 	suite.Require().Equal(&expParams, res.Params)
 }
+
+func (suite *KeeperTestSuite) TestQueryDenomHash() {
+
+	reqTrace := types.DenomTrace{
+		Path:      "transfer/channelToA/transfer/channelToB",
+		BaseDenom: "uatom",
+	}
+
+	var (
+		req     *types.QueryDenomHashRequest
+		expHash = reqTrace.Hash().String()
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"invalid trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: "transfer/channelToA/transfer/",
+				}
+			},
+			false,
+		},
+		{
+			"not found denom trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), reqTrace)
```

Comment:
Ideally the success case doesn't have to perform any functionality. This allows you to be certain the failing cases would normally succeed except in the scenario specified. 

I would set the req to a default before calling tc.malleate in the for loop, I'd also set the denom trace in state. Then the case "not found denom trace" would delete the denom trace in state (or set it to a different trace that doesn't exist in state)

Reply:
@colin-axner @codecov-commenter  I need further clarification on this point. I mistakenly put the request block before and after setting the state. If I removed the duplicate before setting the state would that be good; there is no significant difference between the test method I wrote and the one for GetDemonTrace.

Reply:
Checkout the suggestion I put below. Don't commit that with the github UX because the formatting is probably wrong. But the main difference is adding:

```go
                       req = &types.QueryDenomHashRequest{
					Trace: reqTrace.GetFullDenomPath(),
			}
			suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), reqTrace)
```

in the for loop before tc.malleate()